src/Workspaces/Remote/ServiceHub.CoreComponents/Microsoft.CodeAnalysis.Remote.ServiceHub.CoreComponents.csproj

```diff
@@ -20,27 +21,106 @@
+  <Target Name="LocateCrossgenTargets" DependsOnTargets="Publish">
+    <ItemGroup>
+      <!-- Find all Roslyn assemblies that we want to crossgen -->
+      <_R2RAssemblies Include="@(_ResolvedProjectReferencePaths->'%(FileName)%(Extension)')" />
+      <_R2RAssemblies Include="MessagePack.dll" />
+      <!-- Find all assemblies (including Roslyn and all dependencies) from the actual published location -->
+      <_AllPublishedAssemblyPaths Include="$(PublishDir)\**\*.dll" Exclude="$(PublishDir)\**\*.resources.dll" />
+      <_AllPublishedAssemblies Include="@(_AllPublishedAssemblyPaths->'%(FileName)%(Extension)')" >
+        <_FullFilePath>%(FullPath)</_FullFilePath>
+      </_AllPublishedAssemblies>
+      <!-- Get all non-Roslyn assemblies as a subset of all published ones, we don't crossgen those -->
+      <_NoR2RAssemblies Include="@(_AllPublishedAssemblies)" Exclude="@(_R2RAssemblies)" >
+        <_FullFilePath>%(_AllPublishedAssemblies._FullFilePath)</_FullFilePath>
+      </_NoR2RAssemblies>
+      <_NoR2RAssemblyPaths Include="@(_NoR2RAssemblies->'%(_FullFilePath)')" />
+      <!-- Now we get all Roslyn assemblies in the publish folder -->
+      <_R2RAssemblyPaths Include="@(_AllPublishedAssemblyPaths)" Exclude="@(_NoR2RAssemblyPaths)" />
+    </ItemGroup>
+    
+    <PropertyGroup>
+      <CrossgenWorkFolder>$(PublishDir)\CrossGen\</CrossgenWorkFolder>
+      <OriginalAssemblyFolder>$(CrossgenWorkFolder)OriginalAssemblies\</OriginalAssemblyFolder>
+    </PropertyGroup>   
+
+     <Message Text="Moving Files @(_R2RAssemblyPaths)"/>
+     <Move SourceFiles="@(_R2RAssemblyPaths)" DestinationFolder="$(OriginalAssemblyFolder)">
+       <Output 
+          TaskParameter="DestinationFiles"
+          ItemName="_FilesWritten"/>
+     </Move>
+
+    <ItemGroup>
+      <_CrossgenTargetPaths Include="$(OriginalAssemblyFolder)\*.dll" />
+      <_RoslynDependencies Include="@(_CrossgenTargetPaths)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="CompileReadyToRun" DependsOnTargets="LocateCrossgenTargets;LocateDependencies" Outputs="%(_CrossgenTargetPaths.FullPath)">    
+    <PropertyGroup>    
+      <_Crossgen2Path>$(PkgMicrosoft_NETCore_App_crossgen2_win-x64)\tools\crossgen2.dll</_Crossgen2Path>
+      <_R2ROptimizeAssemblyPath>%(_CrossgenTargetPaths.FullPath)</_R2ROptimizeAssemblyPath>
+      <_R2ROptimizeAssemblyOutputPath>$(PublishDir)\%(_CrossgenTargetPaths.Filename)%(_CrossgenTargetPaths.Extension)</_R2ROptimizeAssemblyOutputPath>      
+      <_RspFilePath>$(CrossgenWorkFolder)%(_CrossgenTargetPaths.Filename).CrossgenArgs.rsp</_RspFilePath>
+    </PropertyGroup>
+    <Delete Files="$(_RspFilePath)" Condition="exists('$(_RspFilePath)')" />
+    <ItemGroup>
+      <RspFile Include="$(_RspFilePath)" />
+      <Lines Include="$(_R2ROptimizeAssemblyPath)" />
+      <Lines Include="--out:$(_R2ROptimizeAssemblyOutputPath)" />
+      <Lines Include="--targetarch:x64" />
```

Comment:
>x64 [](http://example.com/codeflow?start=35&length=3)

What about ARM?

Reply:
We will have a single vsix containing x64 R2R bits for both x64 and arm for now. Net step would be some setup refactoring to have .NET core remote host bits in its own package and then have separate packages targeting x64 and arm